src/main/scala/beam/router/Modes.scala

```diff
@@ -241,3 +251,237 @@ object Modes {
   }
 
 }
+
+object TourModes {
+  import beam.router.Modes.BeamMode
+  import beam.router.Modes.BeamMode._
+
+  sealed abstract class BeamTourMode(
+    val value: String,
+    val vehicleCategory: Seq[VehicleCategory],
+    val allowedBeamModes: Seq[BeamMode],
+    val allowedBeamModesForFirstAndLastLeg: Seq[BeamMode]
+  ) extends StringEnumEntry
+      with ExponentialLazyLogging {
+
+    import BeamTourMode._
+
+    private def getModeFromVehicle(beamVehicle: BeamVehicle): BeamMode = {
+      beamVehicle.beamVehicleType.vehicleCategory match {
+        case VehicleCategory.Car                => CAR
+        case VehicleCategory.Bike               => BIKE
+        case VehicleCategory.Class78Tractor     => FREIGHT
+        case VehicleCategory.Class78Vocational  => FREIGHT
+        case VehicleCategory.Class456Vocational => FREIGHT
+        case VehicleCategory.Class2b3Vocational => FREIGHT
+        case _                                  => WALK
+      }
+    }
+
+    def allowedBeamModesGivenAvailableVehicles(
+      vehicles: Vector[VehicleOrToken],
+      firstOrLastLeg: Boolean
+    ): Seq[BeamMode] = {
+      val relevantModes = if (firstOrLastLeg) { allowedBeamModesForFirstAndLastLeg }
+      else allowedBeamModes
+      if (
+        vehicles
+          .exists { vehOrToken =>
+            !vehOrToken.vehicle.isSharedVehicle && getModeFromVehicle(vehOrToken.vehicle).in(relevantModes)
+          }
+      ) { relevantModes }
+      else { Seq.empty[BeamMode] }
+    }
+
+    def isVehicleBased: Boolean = this match {
+      case WALK_BASED => false
+      case _          => true
+    }
+  }
+
+  object BeamTourMode extends StringEnum[BeamTourMode] with StringCirceEnum[BeamTourMode] with ExponentialLazyLogging {
+
+    override val values: immutable.IndexedSeq[BeamTourMode] = findValues
+
+    def getTourModeAndVehicle(
+      trips: Vector[EmbodiedBeamTrip],
+      availableVehicles: Vector[VehicleOrToken] = Vector.empty[VehicleOrToken],
+      currentTourPersonalVehicle: Option[Id[BeamVehicle]] = None
+    ): mutable.Map[Option[BeamTourMode], mutable.Map[EmbodiedBeamTrip, Option[BeamVehicle]]] = {
```

Comment:
We are returning a map here. The keys of this Map is Option[BeamTourMode]. But we are only putting Some objects there. So Option is not really needed here.